Source/Laser/LaserEnvelope.H

```diff
@@ -0,0 +1,55 @@
+#include <complex>
+#include <iostream>
+#include <complex>
+#include <vector>
+
+
+#include <iostream>
+#include <vector>
+
+class LaserEnvelope {
+private:
+    amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > > A_potential_vector;
```

Comment:
```suggestion
    amrex::Vector< amrex::FabArray< amrex::BaseFab <Complex> > > A_laser_envelope;
```

Reply:
In the end, we decided to use a regular `MultiFab` instead (easier for output) and have two components (for real and imaginary part)

Reply:
This is an example of allocation of a scalar MultiFab, `rho_fp` in this case, with a chosen number of components (by components we mean the fourth index when accessing the array data):
https://github.com/ECP-WarpX/WarpX/blob/8ec6b19767cfaea7949fb47053c788b001994ef8/Source/WarpX.cpp#L2246
The number of components in this case is given by the input argument `rho_ncomps`. In our case we would want to pass `2`, resulting in data access such that `(i,j,k,0)` would represent the real part of the data at cell `(i,j,k)` and `(i,j,k,1)` would represent the imaginary part of the same data.

Reply:
The first step here would be to define a constructor of the new class `LaserEnvelope`, such that it contains at least the allocation of the scalar envelope field (a member variable of the class), which is currently implemented in the separate function `AllocateLaserEnvelope`.